content/docs/user-guide/gems/reference/kythera-ai/spatial-query-system.md

```diff
@@ -0,0 +1,723 @@
+    
+
+#### `Heading`
+
+*   Returns the cosine of the angle between the querier's forward direction and the direction to the current point.
+    
+
+#### `IsCoverOfType(coverType)`
+
+*   Returns true if the current point has the given cover type.
+    
+
+#### **NavRaycast(object, desiredResult)**
+
+*   Performs navmesh raycast from point to object. 
+    
+
+#### **OctreeRaycast(Object, DesiredResult, ShipRadius)**
+
+*   Performs octree raycast from point to Object.
+    
+
+#### **NavDistance(object, maxPathDistance)**
+
+*   Calculates the pathing distance between the current point and the given object. 
+    
+
+#### `OctreeSurfaceDistance(Object)`
+
+*   Calculates the approximate distance between the current point and the nearest octree surface
+    
+    *   Octree parameter "MaxRadius" needs to be high for this to be useful some distance from octree surfaces
+        
+    *   Object parameter should be Querier (it is only used to obtain the octree)
+        
+
+#### **GetBlackboard\[\*\]**
+
+*   Returns a value of specified type dependant on the function listed below, from the entity blackboard under the specified key.
+    
+    *   Named Parameters: 
+        
+        *    `Key (StringHash)`
+            
+    *   Available function types:
+        
+        *   _**GetBlackboardFloat**_(`Key`)
+            
+        *   _**GetBlackboardInt**(_`Key`)
+            
+        *   _**GetBlackboardBool**_(`Key`)
+            
+        *   _**GetBlackboardStringHash**_(`Key`)
+            
+        *   _**GetBlackboardPos**_(`Key`)
+            
+        *   _**GetBlackboardVec3**_(`Key`)
+
+## Debugging spatial queries
+
+### Debugging parser errors
+
+When working with parsed Spatial Queries (either from a .sqs file in Lumberyard, or an SQS Asset in Unreal Engine), you may sometimes make errors when writing your queries. This can lead to errors that cause your spatial query to be unparseable, meaning that Kythera will not be able to use it.
+
+In these situations, we output error logging to the Console:
+
+![SQS errors logged to the console](/images/user-guide/gems/kythera-ai/sqs-console.png)
+
+If you have modified your spatial query and it no longer runs, be sure to check here first.
+
+### Visual debugging
+
+Visual debugging of the spatial query system is available in Kythera. This is enabled on a per-entity basis from within the Kythera Inspector, by setting **Debugging**→**DebugDraw**→**SpatialQuery**→**Enable** to true for that entity:
+
+![Visual debugging of spatial queries with the Kythera AI Inspector](/images/user-guide/gems/kythera-ai/sqs-visual-debugging)
+
+(Note that global debug draw must also be enabled - either by using the **Enable Debug Draw** button in the Lumberyard Editor, or by setting console variable `kyt_DrawMaster` to 1.)
+
+Once this is enabled, debug visualization will be drawn into your scene. 
+
+![Debug draw of spatial queries](/images/user-guide/gems/kythera-ai/sqs-debug-visualization.png)
```

Comment:
Rename the png included with this PR to match this link (`sqs-debug-visualization.png`).